Extensions/dnSpy.AsmEditor/UndoRedo/Commands.cs

```diff
@@ -47,11 +52,15 @@ sealed class UndoRedoCommmandLoader : IAutoLoaded {
 			var count = undoCommandService.Value.NumberOfModifiedDocuments;
 			if (count == 0)
 				return;
-
 			var msg = count == 1 ? dnSpy_AsmEditor_Resources.AskExitUnsavedFile :
 					string.Format(dnSpy_AsmEditor_Resources.AskExitUnsavedFiles, count);
 			var res = messageBoxService.Show(msg, MsgBoxButton.Yes | MsgBoxButton.No);
-			if (res == MsgBoxButton.No || res == MsgBoxButton.None)
+            if (res == MsgBoxButton.Yes) {
+                appWindow.MainWindow.Hide();
```

Comment:
The code gets called in a 'window closing' event. Other code could cancel the closing and then the Window would be minimized which is very confusing.